p3/listeners.py

```diff
@@ -14,42 +14,6 @@
 
 log = logging.getLogger('p3')
 
-def on_order_created(sender, **kwargs):
-    if sender.total() == 0:
```

Comment:
Is it safe to assume this case is never going to be true? removing the rest of this function makes sense.

Reply:
Intentional. It's part of moving from signals to explicit calls in the relevant places in the code.
on_purchase_completed is basically an email notification, and better place for it would be somewhere in the code that actually processes orders. As far as will that if ever be true – yes, in cases where we use full discount codes order.total() could be 0.

Reply:
Gotcha - so if I understand correctly, the `on_purchase_completed` call should be moved to `Order.save` as part of this PR, right? Otherwise we lose this notification.

Reply:
Unless there is no `purchase-complete` template/it hasn't been updated/used in a while. Then it would probably be safe to remove.